### PR TITLE
DT-1796: Log prompt answers for PINT

### PR DESCRIPTION
**Pull request by nt-gt**
This issue is now known as SD-492

### CODE REVIEW

**Comment by nt-gt** [comment]
Hi @jkosternl 

I thought this would solve DT-1796, but one of the manual tests fails with "sandbox" being null:

```

java.lang.NullPointerException: Cannot read field "sandboxName" because "sandbox" is null

	at org.dcsa.conformance.manual.ManualTestBase.resetSandbox(ManualTestBase.java:238)
	at org.dcsa.conformance.manual.ManualScenarioTest.runManualTests(ManualScenarioTest.java:113)
	at org.dcsa.conformance.manual.ManualScenarioTest.lambda$testStandards$2(ManualScenarioTest.java:71)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
	at org.dcsa.conformance.manual.ManualScenarioTest.lambda$testStandards$3(ManualScenarioTest.java:69)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1708)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
	at org.dcsa.conformance.manual.ManualScenarioTest.lambda$testStandards$4(ManualScenarioTest.java:65)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
	at org.dcsa.conformance.manual.ManualScenarioTest.testStandards(ManualScenarioTest.java:61)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:179)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:1024)
	at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:762)
	at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:276)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:1024)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
	at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:276)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1708)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
	at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:276)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1708)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
	at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:276)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1708)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1596)
```

Can you have a look at this and see what I am doing wrong?

(It is the `Standard: PINT - 2nd run: true` that fails with a NPE)

**Review by jkosternl — APPROVED** [review]
Looking good, as far as I understand the framework currently.

File: pint/src/main/java/org/dcsa/conformance/standards/eblinterop/PintScenarioListBuilder.java [hunk 1]
```diff
@@ -65,15 +57,15 @@ public static LinkedHashMap<String, PintScenarioListBuilder> createModuleScenari
...
+              )/*,
               receiverStateSetup(ScenarioClass.FAIL_W_503).then(
                 initiateTransferUnsignedFailure(503, SenderTransmissionClass.VALID_TRANSFER).thenEither(
                   resetScenarioClass(ScenarioClass.NO_ISSUES).then(
                     initiateAndCloseTransferAction(PintResponseCode.RECE)),
                   initiateTransferUnsignedFailure(503, SenderTransmissionClass.VALID_TRANSFER).then(
                     resetScenarioClass(ScenarioClass.NO_ISSUES).then(
                       initiateAndCloseTransferAction(PintResponseCode.RECE)))
-                ))
+                ))*/
```
[the remark this answers is not in the document]
Comment (nt-gt) [reply pint/src/main/java/org/dcsa/conformance/standards/eblinterop/PintScenarioListBuilder.java:68]: Thanks; will review these when I review the scenarios with the POs

**Comment by jkosternl** [comment]
What about @preetamnpr and @palatsangeetha on this PR? Can you both also check it, because you 2 have more experience in writing standards then me. (reviewing can also be done after merging 😉 )

**Comment by preetamnpr** [comment]
I was bit held up in the TNT work. I have reviewed it and it looks fine to me apart from the commented code, for which @nt-gt  have already replied.

**Comment by palatsangeetha** [comment]
I looked at the PR yesterday and it looked good to me except the commented part. Sorry was caught up with ebl issuance conformance issue and slipped my mind to approve the PR.

**Pull request merged**